packages/trpc-interface/src/trpc-caller-link.ts

```diff
@@ -0,0 +1,57 @@
+import { AnyRouter } from "@trpc/server";
+// eslint-disable-next-line import/no-internal-modules
+import { observable } from "@trpc/server/observable";
+import { TRPCClientError, TRPCLink } from "@trpc/client";
+
+type MemoryLinkOptions<TRouter extends AnyRouter> = {
+  appRouter: TRouter;
+  createContext?: () => TRouter["_def"]["_config"]["$types"]["ctx"];
+};
+
+/**
+ * https://github.com/trpc/trpc/issues/3335
+ *
+ * createCaller and createTRPCProxyClient provides different interfaces,
+ * here we provide callerLink which can be used as a [trpc client link](https://trpc.io/docs/links)
+ * Allowing us to call router api without http but through createTRPCProxyClient interface
+ * See trpc-caller-link.test.ts for details
+ **/
+export const callerLink = <TRouter extends AnyRouter>(
```

Comment:
can we call it full name? TRouter what is it? is it TrpcRouter?

Reply:
`TRouter` means `TemplateRouter` its typescript type, its often we add `T` for template parameters, names same as in native links https://github.com/trpc/trpc/blob/main/packages/client/src/links/httpLink.ts#L12